server.go

```diff
@@ -1372,6 +1375,11 @@ func (s *Server) Shutdown() {
 	// TODO(aditya) shut down workers and socket readers
 	s.logger.Info("Shutting down server gracefully")
 	close(s.shutdown)
+	if s.FlushOnShutdown {
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(s.Interval))
```

Comment:
Minor nit, `context.WithTimeout(context.Background(), s.Interval)` is one less step.